core/includes/common.inc

```diff
@@ -1673,11 +1701,14 @@ function flood_is_allowed($name, $threshold, $window = 3600, $identifier = NULL)
  * @param string $name
  *   The name of the object being stored. Must be less than 128 characters.
  *
- * @return
+ * @return string|false|null
```

Comment:
According to PHP docs unserialize() can return anything of bool, int, float, string, array or object.
And it doesn't seem like tempstore_set() limits the type of value in any way... `mixed` ?